File: php/features.php
<?php
declare(strict_types=1);

// WaifuPics SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class WaifuPicsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new WaifuPicsBaseFeature();
            case "test":
                return new WaifuPicsTestFeature();
            default:
                return new WaifuPicsBaseFeature();
        }
    }
}

File: py/features.py
# WaifuPics SDK feature factory

from feature.base_feature import WaifuPicsBaseFeature
from feature.test_feature import WaifuPicsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WaifuPicsBaseFeature(),
        "test": lambda: WaifuPicsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
